requests_futures/sessions.py

```diff
@@ -51,16 +51,19 @@ def __init__(self, executor=None, max_workers=2, session=None, *args,
         * If you provide both `executor` and `max_workers`, the latter is
           ignored and provided executor is used as is.
         """
+        _adapter_kwargs = {}
         super(FuturesSession, self).__init__(*args, **kwargs)
         self._owned_executor = executor is None
         if executor is None:
             executor = ThreadPoolExecutor(max_workers=max_workers)
             # set connection pool size equal to max_workers if needed
             if max_workers > DEFAULT_POOLSIZE:
-                adapter_kwargs = dict(pool_connections=max_workers,
-                                      pool_maxsize=max_workers)
-                self.mount('https://', HTTPAdapter(**adapter_kwargs))
-                self.mount('http://', HTTPAdapter(**adapter_kwargs))
+                _adapter_kwargs.update({'pool_connections': max_workers,
+                                       'pool_maxsize': max_workers})
+
+        _adapter_kwargs.update(adapter_kwargs or {})
+        self.mount('https://', HTTPAdapter(**_adapter_kwargs))
+        self.mount('http://', HTTPAdapter(**_adapter_kwargs))
```

Comment:
Would the change here to always call mount be equivalent to the custom `executor` case where we didn't touch them in the past? Effectively the default with an executor will now be:

```python
self.mount('https://', HTTPAdapter())
self.mount('http://', HTTPAdapter())
```

Reply:
The underlying requests library does the same thing:

https://github.com/requests/requests/blob/master/requests/sessions.py#L397-L398

Supplying an empty `dict` should result in the same behavior. Take for example the following:

```
>>> def foo(bar='baz'):
...     print(bar)
...
>>> foo(**{})
baz
>>>
```

That said, this _could_ change so hid this initialization behind a guard so that it only runs _if_ `adapter_kwargs` was supplied.